src/python/pants/engine/internals/build_files.py

```diff
@@ -38,8 +44,33 @@ def _key_func(entry):
     return key
 
 
+@dataclass(frozen=True)
+class BuildFilePreludeSymbols:
+    symbols: Dict[str, Any]
+
+
 @rule
-async def parse_address_family(address_mapper: AddressMapper, directory: Dir) -> AddressFamily:
+async def evalute_preludes(
+    prelude_option: PreludeOption, build_file_imports_behavior: BuildFileImportsBehaviorWrapper
+) -> BuildFilePreludeSymbols:
+    snapshot = await Get[Snapshot](PathGlobs, PathGlobs(prelude_option.value))
```

Comment:
Should this fail if it cannot find the files?

Also, nit that you can use the inline syntax:

```suggestion
    snapshot = await Get[Snapshot](PathGlobs(prelude_option.value))
```

Reply:
A good question... I can see arguments for either case. For now, I'll leave it as ignore, because it's safer to become more strict than more loose.